R/HS_PGA_num.R

```diff
@@ -0,0 +1,39 @@
+#' Calculates HS-PGA scores based on abscess, draining fistula, and inflammatory nodule counts.
+#'
+#' @param abscess_fistula Integer vector representing the sum of abscess and draining fistula counts.
+#' @param inflammatory_nodule Integer vector representing the inflammatory nodule count.
+#' @param non_inflammatory_nodule Integer vector representing the non-inflammatory nodule count.
+#' @return An integer vector representing the HS-PGA scores (1: Clean, 2: Minimal, 3: Mild, 4: Moderate, 5: Severe, 6: Very Severe).
+#' @export
+#' @examples
+#' 
+#' abscess_fistula <- c(0, 0, 1, 0, 1, 2, 6)
+#' inflammatory_nodule <- c(0, 0, 0, 3, 5, 8, 12)
+#' non_inflammatory_nodule <- c(0, 1, 0, 0, 0, 0, 0)
+#'
+#' hs_pga_num(abscess_fistula, inflammatory_nodule, non_inflammatory_nodule)
+#'
+
+hs_pga_num <- function(abscess_fistula, inflammatory_nodule, non_inflammatory_nodule) {
+  
+  # Assertions using checkmate package
+  checkmate::assert_integerish(abscess_fistula, lower = 0, any.missing = FALSE, null.ok = FALSE)
```

Comment:
There will likely be cases with `NA` input values.  In some cases, these can still give results.  For example, the rule for a score of 6 could be `TRUE` if `inflammatory_nodule` is missing but `abscess_fistula > 5`.  So, please remove the requirement that all values must not be missing.  That will require adaptation of the code below because missing values are not allowed as subscripts.

Reply:
Replaced with #10